tests/feature/feature_console/deregister.pm

```diff
@@ -0,0 +1,64 @@
+# Copyright (C) 2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Test the function of SUSEConnect to deregister a module
+# Maintainer: Yi Xu <yxu@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+
+sub run() {
+    select_console 'root-console';
+
+    # check and list registered modules
+    my $out = script_output "SUSEConnect --status-text";
+    $out =~ /Registered/ or die "There is no module registered";
+
+    my $count = script_output("SUSEConnect --status-text | grep -c \"^\\s*Registered\"", 100);
+    if ($count eq 0) {
+        die "There's no module registered";
+    }
+
+    # check if a certain module is registered, e.g. containers.
+    my $module  = "sle-module-containers";
+    my $abbrv   = "contm";
+    my $version = "12";
+    my $arch    = get_var('ARCH');
+    die "Containers Module needs to be part of SCC_ADDONS for this test" unless check_var_array('SCC_ADDONS', $abbrv);
+    assert_script_run("SUSEConnect --status-text | grep -A 3 $module | grep \"^\\s*Registered\"", 100);
+
+    # deregister the module and check if it is successful
+    assert_script_run "SUSEConnect --de-register --product $module/$version/$arch";
+    assert_script_run("SUSEConnect --status-text | grep -A 3 $module | grep \"Not Registered\"", 100);
+
+    # check if other modules are affected
+    my $count_dereg = script_output("SUSEConnect --status-text | grep -c \"^\\s*Registered\"", 100);
+    if ($count_dereg eq $count - 1) {
+        type_string "One module got deregistered successfully!\n";
+    }
+    else {
+        die "SUSEConnect deregister didn't work properly!";
+    }
+
+    # register the module again, check if it is successful. OR using yast_scc_registration();
+    my $regcode = get_var 'SCC_REGCODE_CONTM';
```

Comment:
still , I don't think here is any containers specific key

correct way to register containers module IS :

```SUSEConnect    -p sle-module-containers/12/x86_64``` on correctly registered SLE12+ system